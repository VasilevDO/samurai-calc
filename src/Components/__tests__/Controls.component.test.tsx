import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import Controls from '../Controls.component';

describe('ControlsComponent', () => {
	const controls = [1, 2, '3', '='];
	const action = jest.fn();

	test('should render a correct element', () => {
		const {container} = render(<Controls controls={controls} clickHandler={action}/>);
		expect(container).toMatchSnapshot();
	});
});

