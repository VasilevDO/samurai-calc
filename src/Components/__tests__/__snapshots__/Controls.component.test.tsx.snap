// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsComponent should render a correct element 1`] = `
.c1 {
  display: table-cell;
  vertical-align: middle;
  height: calc(2em + 8px);
  width: calc(2em + 8px);
  margin: 0 10px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 1em;
  line-height: 1em;
  font-weight: inherit;
  text-align: center;
  border: none;
  color: white;
  background-color: inherit;
}

.c1:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c1:hover {
  background-color: rgba(255,255,255,0.12);
}

.c2 {
  display: table-cell;
  vertical-align: middle;
  height: calc(2em + 8px);
  width: calc(2em + 8px);
  margin: 0 10px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 1em;
  line-height: 1em;
  font-weight: inherit;
  text-align: center;
  border: none;
  color: #2B589A;
  background-color: #FFFFFF;
}

.c2:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c2:hover {
  background-color: rgba(255,255,255,0.7);
}

.c0 {
  display: grid;
  grid-template-columns: repeat(4,auto);
  grid-gap: 16px 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0;
  }
}

<div>
  <div
    class="c0"
    data-testid="controls"
  >
    <button
      class="c1"
    >
      1
    </button>
    <button
      class="c1"
    >
      2
    </button>
    <button
      class="c1"
    >
      3
    </button>
    <button
      class="c2"
    >
      =
    </button>
  </div>
</div>
`;
